boards/arm/nrf5340pdk_nrf5340/nrf5340pdk_nrf5340_cpuapp_partition_conf.dts

```diff
@@ -44,10 +44,6 @@
  * - Upper 64 kB SRAM allocated as Shared memory (sram0_shared)
  *   (see nrf5340pdk_nrf5340_shared_sram_planning_conf.dts)
  */
-&sram0 {
```

Comment:
@hakonfam the problem here is that the reduction of sram0 is done to leave space for shared memory.
In other words, the mess here is that we treat sram0 not as physical ram node, but as a system"partition" 
Ideally we should have dts nodes that do not resize because they represent physical memory and nodes that are partitions and can be resized by overlays

Reply:
I don't think I understand what you mean. I'm trying to solve the issue where `sram0` is being re-defined to something other than the complete size of the RAM in the SoC. Whenever some application or feature need to have a ram partition reserved, it must use something other than `sram0`. In the case of the nrf5340 the secure code would use `sram0_s` and the non-secure code would use `sram0_ns`

Reply:
The reason for re-sizing sram0 here is that we make room for the shared-RAM between Application and Network MCU. It has nothing to do with security.

(Speaking about security: there's also the case of a default-secure firmware which should be using the whole ram)

Other than that, I agree that we need to _separate_ how we describe physical RAM areas and image RAM areas (e.g. partitions, shared memory areas, etc.) in Zephyr.

But as I said above, this PR cannot go in as-is because of the shared-ram issue

Reply:
I understand that we change reduce `sram0` to reserve memory for some other purpose (in this case shared memory area). I don't see how this affects the proposed solution. The shared memory is defined by a separate node.